src/Controller/BlogPostController.php

```diff
@@ -0,0 +1,207 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\DTO\BlogPostEnquiry;
+use App\DTO\BlogPostListEnquiry;
+use App\Entity\BlogPost;
+use App\Entity\BlogUser;
+use App\Repository\BlogUserRepository;
+use App\Service\BlogPostService;
+use App\Service\Serializer\DTOSerializer;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Exception\NotSupported;
+use Doctrine\ORM\OptimisticLockException;
+use Doctrine\ORM\ORMException;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use App\Repository\BlogPostRepository;
+
+class BlogPostController extends AbstractController
+{
+
+    /**
+     * @Route("/blogposts/{userID?}", name="blog-posts", methods={"GET"})
+     * @throws NotSupported
+     * @throws \JsonException
+     *
+     * returns blogposts by userID or If userID null, all blogposts
+     */
+    public function getBlogPostList(
+        Request $request,
+        EntityManagerInterface $entityManager,
+        DTOSerializer $serializer,
+        ?int $userID = null,
+    ):
+    Response {
+        /** @var BlogPostListEnquiry $enquiry */
+        $enquiry = $serializer->deserialize(
+            $request->getContent(),
+            BlogPostListEnquiry::class,
+            'json'
+        );
```

Comment:
[Tony]
- look into single key access to data from body, ie `get['key']`


[Jeff]
- `GET` doesn't usually have request body, use query params